signer/tests/integration/withdrawal_reject.rs

```diff
@@ -0,0 +1,432 @@
+        deployer: StacksAddress::burn_address(false),
+    };
+
+    (complete_withdrawal_tx, req_ctx)
+}
+
+/// For this test we check that the `RejectWithdrawalV1::validate` function
+/// returns okay when everything matches the way that it is supposed to.
+#[tokio::test]
+async fn reject_withdrawal_validation_happy_path() {
+    // Normal: this generates the blockchain as well as a transaction
+    // sweeping out the funds for a withdrawal request. This is just setup
+    // and should be essentially the same between tests.
+    let db = testing::storage::new_test_database().await;
+    let mut rng = rand::rngs::StdRng::seed_from_u64(51);
+    let (rpc, faucet) = regtest::initialize_blockchain();
+
+    let amount = 1_000_000;
+    let test_signer_set = TestSignerSet::new(&mut rng);
+    let deposit_amounts = DepositAmounts { amount, max_fee: amount / 2 };
+    let mut setup =
+        TestSweepSetup2::new_setup(test_signer_set.clone(), &faucet, &[deposit_amounts]);
+
+    eprintln!("sweep block hash {:?}", &setup.sweep_block_hash());
+    eprintln!("sweep block height {:?}", &setup.sweep_block_height());
+
+    setup.submit_sweep_tx(rpc, faucet, true);
+
+    eprintln!("sweep block hash {:?}", &setup.sweep_block_hash());
+    eprintln!("sweep block height {:?}", &setup.sweep_block_height());
+
+    let ctx = TestContext::builder()
+        .with_storage(db.clone())
+        .with_first_bitcoin_core_client()
+        .with_mocked_stacks_client()
+        .with_mocked_emily_client()
+        .build();
+
+    let public_keys = test_signer_set
+        .keys
+        .iter()
+        .cloned()
+        .collect::<BTreeSet<PublicKey>>();
+    ctx.state().update_current_signer_set(public_keys);
+
+    // Normal: the signer follows the bitcoin blockchain and event observer
+    // should be getting new block events from bitcoin-core. We haven't
+    // hooked up our block observer, so we need to manually update the
+    // database with new bitcoin block headers.
+    backfill_bitcoin_blocks(&db, rpc, &setup.sweep_block_hash().unwrap()).await;
+
+    // Normal: we take the sweep transaction as is from the test setup and
+    // store it in the database.
+    setup.store_sweep_tx(&db).await;
+
+    // Normal: we need to store a row in the dkg_shares table so that we
+    // have a record of the scriptPubKey that the signers control.
+    setup.store_dkg_shares(&db).await;
+
+    // Normal: the request and how the signers voted needs to be added to
+    // the database. Here the bitmap in the withdrawal request object
+    // corresponds to how the signers voted.
+    setup.reject_withdrawal_request();
+    setup.store_withdrawal_request(&db).await;
+    setup.store_withdrawal_decisions(&db).await;
+
+    // Generate the transaction and corresponding request context.
+    let (reject_withdrawal_tx, req_ctx) = make_withdrawal_reject2(&setup);
+
+    // Generate more blocks then backfill the DB
+    let mut hashes = faucet.generate_blocks(6);
```

Comment:
Ahh, I see why you want to get the chain tip during validation. Instead you should give `make_withdrawal_reject2` the chain tip.

Reply:
I passed it `db` so the tests which used `make_withdrawal_reject2` wouldn't have to all find the chain tip.